gitops-generator/pkg/generate/generate.go

```diff
@@ -0,0 +1,173 @@
+package generate
+
+import (
+	"context"
+	"fmt"
+
+	appstudiov1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	gitops "github.com/redhat-appstudio/application-service/gitops-generator/pkg/gitops"
+	"github.com/redhat-appstudio/application-service/gitops-generator/pkg/gitops/prepare"
+	"github.com/redhat-appstudio/application-service/pkg/util"
+	gitopsgenv1alpha1 "github.com/redhat-developer/gitops-generator/api/v1alpha1"
+	gitopsgen "github.com/redhat-developer/gitops-generator/pkg"
+	"github.com/redhat-developer/gitops-generator/pkg/util/ioutils"
+	"github.com/spf13/afero"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	client "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type GitOpsGenParams struct {
+	Generator gitopsgen.Generator
+	RemoteURL string
+	Branch    string
+	Context   string
+	Token     string
+}
+
+func GenerateGitopsBase(ctx context.Context, client client.Client, component appstudiov1alpha1.Component, appFs afero.Afero, gitopsParams GitOpsGenParams) error {
```

Comment:
arent these funcs in the library? I dont see it on main anymore

Reply:
The code from these is copied from the Component and ASEB (For GenerateGitopsOverlays) controllers. Basically the wrapper/setup code needed before calling the gitops generator library.